### PR TITLE
feat: add blob sidecar index check

### PR DESCRIPTION
**Pull request by ensi321**
Adding a validation rule we've missed when validating blob sidecar:
https://github.com/ethereum/consensus-specs/blob/5567434a77a87b53cb9b2e2c8812220fc5e9ee84/specs/deneb/p2p-interface.md?plain=1#L179

### CODE REVIEW

**Review by nflaig — APPROVED** [review]
Good catch, although seems bit redundant considering the subnet check below but I guess that's how the spec states it.

File: packages/beacon-node/src/chain/validation/blobSidecar.ts [hunk 1]
```diff
@@ -18,6 +18,15 @@ export async function validateGossipBlobSidecar(
 ): Promise<void> {
   const blobSlot = blobSidecar.signedBlockHeader.message.slot;
 
+  // [REJECT] The sidecar's index is consistent with `MAX_BLOBS_PER_BLOCK` -- i.e. `blob_sidecar.index < MAX_BLOBS_PER_BLOCK`.
+  if (blobSidecar.index < chain.config.MAX_BLOBS_PER_BLOCK) {
```
Comment (nflaig) [inline packages/beacon-node/src/chain/validation/blobSidecar.ts:22]: just noticed the check is incorrect, this should be 
```ts
blobSidecar.index >= chain.config.MAX_BLOBS_PER_BLOCK
```

Comment (ensi321) [reply packages/beacon-node/src/chain/validation/blobSidecar.ts:22]: Oops gonna open another PR

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.24.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.24.0) :tada:

**Pull request merged**